libraries/localization-utilities/src/TypingsGenerator.ts

```diff
@@ -62,7 +89,31 @@ export class TypingsGenerator extends StringValuesTypingsGenerator {
           });
         }
 
-        return { typings };
+        let exportAsDefaultInterfaceName: string | undefined;
+        if (inferDefaultExportInterfaceNameFromFilename) {
+          const lastSlashIndex: number = Math.max(filePath.lastIndexOf('/'), filePath.lastIndexOf('\\'));
+          const extensionIndex: number = Math.max(
+            filePath.lastIndexOf('.resx'),
+            filePath.lastIndexOf('.resjson'),
+            filePath.lastIndexOf('.loc.json')
+          );
+          const fileNameWithoutExtension: string = filePath.substring(lastSlashIndex + 1, extensionIndex);
+          const normalizedFileName: string = fileNameWithoutExtension.replace(/[^a-zA-Z0-9]/g, '');
+          const [firstCharacter, ...restOfCharacters] = normalizedFileName;
+          exportAsDefaultInterfaceName = `I${firstCharacter.toUpperCase()}${restOfCharacters.join('')}`;
```

Comment:
Please use `.charAt(0)` and `.slice(1)`. Destructuring is a spectacularly expensive way to do what is otherwise a trivial operation.